src/services/engine_manager.cpp

```diff
@@ -111,9 +114,21 @@ namespace otto::services {
         case ScreenEnum::synth: [[fallthrough]];
         case ScreenEnum::synth_envelope: [[fallthrough]];
         case ScreenEnum::synth_selector: [[fallthrough]];
-        case ScreenEnum::voices:
-          if (ui_manager.state.active_channel != +ChannelEnum::internal)
+        case ScreenEnum::voices: {
+          if (ui_manager.state.active_channel != +ChannelEnum::internal) {
             ui_manager.state.active_channel = +ChannelEnum::internal;
```

Comment:
this check should not be needed - if an `on_change` signal fires here when it shouldn't, check `new_val != old_val` in that signal handler.

Reply:
I am a bit unsure of what you mean. I think you write this ;)